### ISSUE TITLE
Tracer defaultTags behavior and documentation

### ISSUE THREAD

**Issue by alexgervais** [openzipkin/zipkin-js#330]
Using `zipkin-js` 0.16.0, I tried to understand how the Tracer `defaultTags` property by looking at the original PR and test cases.

Given the following code:
```
	const ctxImpl = new (require('zipkin').ExplicitContext)();
	const recorder = new (require('zipkin').BatchRecorder)({
		logger: new (require('zipkin-transport-http')).HttpLogger({
			endpoint: `${ZIPKIN_BASE_URL}/api/v2/spans`,
			jsonEncoder: require('zipkin').jsonEncoder.JSON_V2
		})
	});

	const tracer = new (require('zipkin').Tracer)({
		ctxImpl: ctxImpl,
		recorder: recorder,
		localServiceName: 'my-app',
		defaultTags: {
			version: '1.0.0'
		}
	});
```

Spans are sent and collected by OpenZipkin. However, they are all missing the expected `version: 1.0.0` tag.

Unable to find the related documentation for Tracer configuration, I'm not sure if my usage if right and this is a bug, or I'm simply not using the feature correctly.

**Comment by jcchavezs** [comment]
Indeed spans should be tagged with default tags and they should be added on span creation. I guess the issue is in https://github.com/openzipkin/zipkin-js/pull/302/files#diff-ac3b9e6d0bce19fd854267694cd42421R108 because in the test we add the tag instead of expect the tracer to automatically add it. Am I wrong @ghermeto ?

**Comment by ghermeto** [comment]
@jcchavezs AFAIK it works just fine:

https://github.com/openzipkin/zipkin-js/blob/master/packages/zipkin/test/trace.test.js#L122-L142

**Comment by jcchavezs** [comment]
@alexgervais would you raise a failing test case for this?

**Comment by alexgervais** [comment]
I created a failing test case: https://github.com/openzipkin/zipkin-js/pull/333
Again, it might just be my understanding of the feature that's not in line with the implemented behavior.

**Comment by ghermeto** [comment]
I added a comment on the PR. The constructor will record the `defaultTags` first.

@jcchavezs  I will add a test to find out why @alexgervais is unable to find the tags in `openZipkin`.

**Comment by ghermeto** [comment]
Ok, definitely a **bug** @alexgervais.

The problem is precisely because `setTags` is being called before the `traceId` is set. In this case, `BatchRecorder` will update a different partial span.

@jcchavezs there are several ways to fix this, but require to move the recording outside of the tracer constructor. My suggestion is to move the recording of the `defaultTags` back to `httpServer` and `httpClient`, but we can keep the defaultTags definition on the tracer, but only record it after the `rpc` is recorded. This way we keep the contract (not a breaking change) and I also know @adriancole didn't want the `defaultTags` in the httpServer and httpClient constructor. Maybe just recoding then would be acceptable?

**Comment by adriancole** [comment]
considering we still want default tags to be on all spans (afaik) then we wouldn't want to move it to the http abstraction. Maybe move to batch recorder, or defer on serviceName being set? Either way, it isn't nice right now, but api break isn't a great idea yet.

**Comment by ghermeto** [comment]
@adriancole my suggestion to fix the issue is #334

In the meantime directly call `Tracer.setTags` method is a workaround for current version.

**Comment by jcchavezs** [comment]
Closed by #334
